src/nplinker/genomics/genomics.py

```diff
@@ -1,44 +1,100 @@
-# Copyright 2021 The NPLinker Authors
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
 from __future__ import annotations
 import csv
-import os
 from os import PathLike
 from pathlib import Path
-import re
-from Bio import SeqIO
+from deprecated import deprecated
 from nplinker.logconfig import LogConfig
 from nplinker.strain_collection import StrainCollection
 from .bgc import BGC
 from .gcf import GCF
-from os import PathLike
-import os
-from pathlib import Path
-
 
 
 logger = LogConfig.getLogger(__name__)
 
-CLUSTER_REGION_REGEX = re.compile('(.+?)\\.(cluster|region)(\\d+).gbk$')
+def map_strain_to_bgc(strains: StrainCollection, bgcs: list[BGC]):
+    """To set BGC object's strain with representative strain object.
+
+    This method changes the list `bgcs` in place.
+
+    It's assumed that BGC.bgc_id is used as strain's name or alias. Then it's
+    possible to lookup the representative strain in the StrainCollection object.
+
+    Args:
+        strains(StrainCollection): A collection of all strain objects.
+        bgcs(list[BGC]): A list of BGC objects.
+
+    Raises:
+        KeyError: Strain id not found in the strain collection.
+    """
+    for bgc in bgcs:
+        try:
+            # assume that bgc.bgc_id is a strain name or alias
+            strain = strains.lookup(bgc.bgc_id)
+        except KeyError as e:
+            raise KeyError(
+                f"Strain id {bgc.bgc_id} from BGC object {bgc.bgc_id} "
+                "not found in the StrainCollection object.") from e
+        bgc.strain = strain
+
+
+def map_bgc_to_gcf(bgcs: list[BGC], gcfs: list[GCF]):
+    """To add BGC objects to GCF object based on GCF's BGC ids.
+
+    This method changes the lists `bgcs` and `gcfs` in place.
+
+    Args:
+        bgcs(list[BGC]): A list of BGC objects.
+        gcfs(list[GCF]): A list of GCF objects.
+
+    Raises:
+        KeyError: BGC id not found in the list of BGC objects.
+    """
+    bgc_dict = {bgc.bgc_id: bgc for bgc in bgcs}
+    for gcf in gcfs:
+        for bgc_id in gcf.bgc_ids:
+            try:
+                bgc = bgc_dict[bgc_id]
+            except KeyError as e:
+                raise KeyError(f"BGC id {bgc_id} from GCF object {gcf.gcf_id} "
+                               "not found in the list of BGC objects.") from e
+            gcf.add_bgc(bgc)
```

Comment:
This looks good - I will introduce a similar function for the molecular families, as pretty much exactly the same is required on the metabolomics side.